functions/src/handlers/CreateEventHandler.ts

```diff
@@ -0,0 +1,116 @@
+        .then(() => {
+            return res.status(200).send({message: 'Created event.'});
+        })
+        .catch((err) => {
+            err.whereInApi = 'CreateEventHandler/addEvent';
+            return res.status(406).send({message: 'Something went wrong', error: err});
+        });
+    }
+
+    function verifyClass (classUid, eventData) {
+        admin.database().ref(`universities/${req.body.university}/${classUid}`)
+        .once('value')
+        .then((classSnap) => {
+            const classData = classSnap.val();
+            if (classData) addEvent(classUid, eventData);
+            else return res.status(404).send({message: 'Class not found.'})
+        })
+        .catch((err) => {
+            err.whereInApi = 'CreateEventHandler/verifyClass';
+            return res.status(406).send({message: 'Something went wrong', error: err});
+        });
+    }
+
+    function getInstructorName(classUid, eventData) {
+        const { instructorUid } = eventData;
+        admin.auth().getUser(instructorUid)
+        .then((user) => {
+            if (user) {
+                eventData.instructorName = user.displayName;
+                verifyClass(classUid, eventData);
+            } else return res.status(404).send({message: 'Couldn\'t get user data.'});
+        })
+        .catch((err) => {
+            err.whereInApi = 'CreateEventHandler/getInstructorName';
+            return res.status(406).send({message: 'Something went wrong', error: err});
+        });
+    }
+
+    return cors(req, res, () => {
+        const { university, classUid, uid, eventData } = req.body;
+        if (university && classUid && uid) {
+            admin.database().ref(`users/${uid}`)
+            .once('value')
+            .then((userSnap) => {
+                const userData = userSnap.val();
+                if (userData) {
+                    if (userData.type === 'teacher' && (userData.universities && (userData.universities[university].indexOf(classUid) !== -1)) ) {
+                        if (validateData(eventData)) {
+                            eventData.instructorUid = '';
+                            eventData.instructorUid = uid;
+                            getInstructorName(classUid, eventData);
+                        } else {
+                            return res.status(400).send({message: 'Missing event data'});
+                        }
+                    } else {
+                        return res.status(403).send({message: 'Not authorized.'});
+                    }
+                } else {
+                    return res.status(404).send({message: 'User not found'});
+                }
+            })
+            .catch((err) => {
+                err.whereInApi = 'CreateEventHandler/cors';
+                return res.status(406).send({message: 'Something went wrong', error: err});
+            });
+        } else {
+            return res.status(400).send({message: 'Missing data'});
+        }
+    });
+}
```

Comment:
new line

Reply:
doesn't matter, ts gets transpiled to js and the builder handles the new line

Reply:
When it gets loaded on someones computer that does support the proper line endings it will result in an unintended commit when their text editor automatically fixes it.

Reply:
ts lint rules given by firebase don't complain. this doesn't follow the same linting as our website folder